File: forum-api/DataAccess/DataObjects/Topic.cs
using System;
using System.Collections.Generic;

namespace forum_api.DataAccess.DataObjects
{
    public partial class Topic
    {
        public Topic()
        {
            Comments = new HashSet<Comment>();
        }

        public int Id { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime? ModificationDate { get; set; }
        public string Title { get; set; } = null!;
        public string Author { get; set; } = null!;

        public virtual ICollection<Comment> Comments { get; set; }
    }
}

File: forum-front/src/models/topic.ts
import { Comment } from "./comment";

export class Topic{

    public Id : number;
    public CreationDate : Date;
    public ModificationDate : Date;
    public Title : string;
    public Author : string;

    public Comments : Comment[];
    
    constructor(id : number, creationDate : Date, modificationDate : Date, title : string, author : string, comments : Comment[]){
        this.Id = id;
        this.CreationDate = creationDate;
        this.ModificationDate = modificationDate;
        this.Title = title;
        this.Author = author;
        this.Comments = comments;
    }
}